indykite_sdk_node/src/sdk/knowledge.ts

```diff
@@ -57,4 +70,195 @@ export class IdentityKnowledgeClient {
       });
     });
   }
+
+  /**
+   * @since 0.4.2
+   */
+  private parseSingleNodeFromPaths(paths: Path[]): Node {
+    if (!paths?.length || !paths[0].nodes?.length) {
+      throw new SdkError(
+        SdkErrorCode.SDK_CODE_3,
+        SkdErrorText.SDK_CODE_3(IdentityKnowledgeClient.name),
+      );
+    }
+    return paths[0].nodes[0];
+  }
+
+  /**
+   * @since 0.4.2
+   */
+  private parseMultipleNodesFromPaths(paths: Path[]): Node[] {
+    if (paths){
+      return paths.reduce((acc, path)=>{
+        if (path?.nodes) {
+          return acc.concat(path.nodes);
+        }else {
+          return acc;
+        }
```

Comment:
```suggestion
        path?.nodes ? acc.concat(path.nodes) : acc
```

This does the same right? that is much shorter and cleaner imo.